defi/src/storeTvlInterval/computeTVL.ts

```diff
@@ -1,98 +1,110 @@
 import BigNumber from "bignumber.js";
 import fetch from "node-fetch";
-import { addStaleCoin, StaleCoins } from "./staleCoins";
-
 const ethereumAddress = "0x0000000000000000000000000000000000000000";
 const weth = "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2";
-const DAY = 24 * 3600;
+
 type Balances = {
   [symbol: string]: number;
 };
 
-export default async function (balances: { [address: string]: string }, timestamp: "now" | number, staleCoins:StaleCoins) {
+function normalizeBalances(balances: { [address: string]: string }) {
+  Object.keys(balances).map((key) => {
+    if (+balances[key] === 0) {
+      delete balances[key];
+      return;
+    }
+
+    const normalisedKey = key.startsWith("0x")
+      ? `ethereum:${key.toLowerCase()}`
+      : !key.includes(":")
+      ? `coingecko:${key.toLowerCase()}`
+      : key;
+
+    // sols case sensitive so no normalising
+    if (key == normalisedKey || key.startsWith("solana:")) return;
+
+    balances[normalisedKey] =
+      normalisedKey in balances
+        ? (Number(balances[normalisedKey]) + Number(balances[key])).toString()
```

Comment:
99% sure this is safe, but if we do `sdk.utils.sumSingleBalance(balances, normalisedKey, balances[key])` it uses bigNumber.js and takes care of edge cases/possible underflow/overflow?

Reply:
one issue is that some balances might be a coingecko id like
{
 bitcoin: 1.2
}

and `sumSingleBalance` runs `toFixed(0)` so what could happen is that 1.2 btc becomes 1 btc (we cut the decimals)